Terraform crashed

### Crash Output

```
panic: value is marked, so must be unmarked first

goroutine 2820 [running]:
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
        /Users/distiller/go/pkg/mod/github.com/zclconf/go-cty@v1.9.0/cty/marks.go:141
github.com/zclconf/go-cty/cty.Value.AsString(0x3749018, 0xc00005c3da, 0x3030f20, 0xc003a02798, 0x32cf900, 0x2)
        /Users/distiller/go/pkg/mod/github.com/zclconf/go-cty@v1.9.0/cty/value_ops.go:1254 +0x4f
github.com/hashicorp/terraform/internal/command/format.ObjectValueID(0x37490f8, 0xc0017d1180, 0x2f418e0, 0xc002c5a360, 0x2e6c4e0, 0xc0017d14f0, 0x2e6c4e0, 0xc0017d1500)
        /Users/distiller/project/project/internal/command/format/object_id.go:34 +0x24b
github.com/hashicorp/terraform/internal/command/format.ObjectValueIDOrName(0x37490f8, 0xc0017d1180, 0x2f418e0, 0xc002c5a360, 0xc000464997, 0x7, 0xc0004649b0, 0x8)
        /Users/distiller/project/project/internal/command/format/object_id.go:117 +0x4d
github.com/hashicorp/terraform/internal/command/views.(*UiHook).PreApply(0xc000430810, 0xc0012d3fe0, 0x1, 0x1, 0x44, 0xc000464997, 0x7, 0xc0004649b0, 0x8, 0x0, ...)
        /Users/distiller/project/project/internal/command/views/hook_ui.go:78 +0x16e
github.com/hashicorp/terraform/internal/terraform.(*NodeAbstractResourceInstance).applyDataSource.func1(0x376e310, 0xc000430810, 0xc003a02800, 0x0, 0x0)
        /Users/distiller/project/project/internal/terraform/node_resource_abstract_instance.go:1651 +0x13e
github.com/hashicorp/terraform/internal/terraform.(*BuiltinEvalContext).Hook(0xc00147b260, 0xc003a02828, 0x0, 0x0)
        /Users/distiller/project/project/internal/terraform/eval_context_builtin.go:100 +0x7c
github.com/hashicorp/terraform/internal/terraform.(*NodeAbstractResourceInstance).applyDataSource(0xc002722e00, 0x3779820, 0xc00147b260, 0xc002e3a640, 0x1, 0x1, 0x0, 0x0)
        /Users/distiller/project/project/internal/terraform/node_resource_abstract_instance.go:1650 +0x249
github.com/hashicorp/terraform/internal/terraform.(*NodeApplyableResourceInstance).dataResourceExecute(0xc001a34900, 0x3779820, 0xc00147b260, 0x18, 0xc001d35428, 0x0)
        /Users/distiller/project/project/internal/terraform/node_resource_apply_instance.go:163 +0x2be
github.com/hashicorp/terraform/internal/terraform.(*NodeApplyableResourceInstance).Execute(0xc001a34900, 0x3779820, 0xc00147b260, 0xc0000ae001, 0xc00141fce8, 0x100d305, 0x2f8ae40)
        /Users/distiller/project/project/internal/terraform/node_resource_apply_instance.go:137 +0x116
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc002bd1110, 0x3779820, 0xc00147b260, 0x345231c8, 0xc001a34900, 0x0, 0x0, 0x0)
        /Users/distiller/project/project/internal/terraform/graph_walk_context.go:129 +0xbf
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1(0x32ab2a0, 0xc001a34900, 0x0, 0x0, 0x0)
        /Users/distiller/project/project/internal/terraform/graph.go:59 +0xbd3
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc000c6bf80, 0x32ab2a0, 0xc001a34900, 0xc002d7dd40)
        /Users/distiller/project/project/internal/dag/walk.go:381 +0x288
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update
        /Users/distiller/project/project/internal/dag/walk.go:304 +0x1246




!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.734424951.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.734424951.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
### Terraform Version
```
Terraform v1.0.4
on darwin_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.54.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
```

```terraform
terraform apply -auto-approve
```

--- a/internal/command/format/object_id.go
+++ b/internal/command/format/object_id.go
@@ -1,6 +1,7 @@
 package format
 
 import (
+	"github.com/hashicorp/terraform/internal/lang/marks"
 	"github.com/zclconf/go-cty/cty"
 )
 
@@ -30,6 +31,11 @@ func ObjectValueID(obj cty.Value) (k, v string) {
 
 	case atys["id"] == cty.String:
 		v := obj.GetAttr("id")
+		if v.HasMark(marks.Sensitive) {
+			break
+		}
+		v, _ = v.Unmark()
+
 		if v.IsKnown() && !v.IsNull() {
 			return "id", v.AsString()
 		}
@@ -38,6 +44,11 @@ func ObjectValueID(obj cty.Value) (k, v string) {
 		// "name" isn't always globally unique, but if there isn't also an
 		// "id" then it _often_ is, in practice.
 		v := obj.GetAttr("name")
+		if v.HasMark(marks.Sensitive) {
+			break
+		}
+		v, _ = v.Unmark()
+
 		if v.IsKnown() && !v.IsNull() {
 			return "name", v.AsString()
 		}
@@ -77,25 +88,41 @@ func ObjectValueName(obj cty.Value) (k, v string) {
 
 	case atys["name"] == cty.String:
 		v := obj.GetAttr("name")
+		if v.HasMark(marks.Sensitive) {
+			break
+		}
+		v, _ = v.Unmark()
+
 		if v.IsKnown() && !v.IsNull() {
 			return "name", v.AsString()
 		}
 
 	case atys["tags"].IsMapType() && atys["tags"].ElementType() == cty.String:
 		tags := obj.GetAttr("tags")
-		if tags.IsNull() || !tags.IsWhollyKnown() {
+		if tags.IsNull() || !tags.IsWhollyKnown() || tags.HasMark(marks.Sensitive) {
 			break
 		}
+		tags, _ = tags.Unmark()
 
 		switch {
 		case tags.HasIndex(cty.StringVal("name")).RawEquals(cty.True):
 			v := tags.Index(cty.StringVal("name"))
+			if v.HasMark(marks.Sensitive) {
+				break
+			}
+			v, _ = v.Unmark()
+
 			if v.IsKnown() && !v.IsNull() {
 				return "tags.name", v.AsString()
 			}
 		case tags.HasIndex(cty.StringVal("Name")).RawEquals(cty.True):
 			// AWS-style naming convention
 			v := tags.Index(cty.StringVal("Name"))
+			if v.HasMark(marks.Sensitive) {
+				break
+			}
+			v, _ = v.Unmark()
+
 			if v.IsKnown() && !v.IsNull() {
 				return "tags.Name", v.AsString()
 			}

--- a/internal/command/format/object_id_test.go
+++ b/internal/command/format/object_id_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/hashicorp/terraform/internal/lang/marks"
 	"github.com/zclconf/go-cty/cty"
 )
 
@@ -56,6 +57,14 @@ func TestObjectValueIDOrName(t *testing.T) {
 			[...]string{"name", "awesome-foo"},
 			[...]string{"name", "awesome-foo"},
 			[...]string{"name", "awesome-foo"},
+		},
+		{
+			cty.ObjectVal(map[string]cty.Value{
+				"name": cty.StringVal("awesome-foo").Mark(marks.Sensitive),
+			}),
+			[...]string{"", ""},
+			[...]string{"", ""},
+			[...]string{"", ""},
 		},
 		{
 			cty.ObjectVal(map[string]cty.Value{
@@ -155,6 +164,16 @@ func TestObjectValueIDOrName(t *testing.T) {
 			cty.ObjectVal(map[string]cty.Value{
 				"tags": cty.MapVal(map[string]cty.Value{
 					"Name": cty.UnknownVal(cty.String),
+				}),
+			}),
+			[...]string{"", ""},
+			[...]string{"", ""},
+			[...]string{"", ""},
+		},
+		{
+			cty.ObjectVal(map[string]cty.Value{
+				"tags": cty.MapVal(map[string]cty.Value{
+					"Name": cty.UnknownVal(cty.String).Mark(marks.Sensitive),
 				}),
 			}),
 			[...]string{"", ""},

